### PR TITLE
Remove unused functions from function map.

### PR DESCRIPTION
**Pull request by yizhang4321**

### CODE REVIEW

File: pkg/agent/baker.go [hunk 1]
```diff
@@ -278,42 +263,9 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
...
-			return getProbes(ports)
-		},
-		"GetSecurityRules": func(ports []int) string {
-			return getSecurityRules(ports)
-		},
-		"GetUniqueNameSuffix": func() string {
-			return cs.Properties.GetClusterID()
-		},
-		"GetVNETAddressPrefixes": func() string {
-			return getVNETAddressPrefixes(cs.Properties)
```
Comment (xuto2) [inline pkg/agent/baker.go:303]: can we remove this function as well

File: pkg/agent/baker.go [hunk 2]
```diff
@@ -278,42 +263,9 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
...
-			return cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision()
-		},
 		"UseManagedIdentity": func() bool {
 			return cs.Properties.OrchestratorProfile.KubernetesConfig.UseManagedIdentity
 		},
-		"GetVNETSubnetDependencies": func() string {
-			return getVNETSubnetDependencies(cs.Properties)
-		},
-		"GetLBRules": func(name string, ports []int) string {
-			return getLBRules(name, ports)
```
Comment (xuto2) [inline pkg/agent/baker.go:291]: remove this function as well

File: pkg/agent/baker.go [hunk 3]
```diff
@@ -278,42 +263,9 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
...
-		},
-		"GetProbes": func(ports []int) string {
-			return getProbes(ports)
```
Comment (xuto2) [inline pkg/agent/baker.go:294]: remove getProbles function

File: pkg/agent/baker.go [hunk 4]
```diff
@@ -278,42 +263,9 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
...
-		},
-		"GetSecurityRules": func(ports []int) string {
-			return getSecurityRules(ports)
```
Comment (xuto2) [inline pkg/agent/baker.go:297]: this

File: pkg/agent/baker.go [hunk 5]
```diff
@@ -278,42 +263,9 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
...
-		},
-		"GetUniqueNameSuffix": func() string {
-			return cs.Properties.GetClusterID()
-		},
-		"GetVNETAddressPrefixes": func() string {
-			return getVNETAddressPrefixes(cs.Properties)
-		},
-		"GetVNETSubnets": func(addNSG bool) string {
-			return getVNETSubnets(cs.Properties, addNSG)
```
Comment (xuto2) [inline pkg/agent/baker.go:306]: this

File: pkg/agent/baker.go [hunk 6]
```diff
@@ -278,42 +263,9 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
...
-		},
-		"GetDataDisks": func(profile *datamodel.AgentPoolProfile) string {
-			return getDataDisks(profile)
```
Comment (xuto2) [inline pkg/agent/baker.go:309]: this

File: pkg/agent/baker.go [hunk 7]
```diff
@@ -216,9 +210,6 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
 		"IsKubernetesVersionGe": func(version string) bool {
```
Comment (xuto2) [inline pkg/agent/baker.go:210]: this is util method, I prefer we keeping it so we can use it freely

File: pkg/agent/baker.go [hunk 8]
```diff
@@ -216,9 +210,6 @@ func getContainerServiceFuncMap(config *NodeBootstrappingConfiguration) template
...
 			return cs.Properties.OrchestratorProfile.IsKubernetes() && IsKubernetesVersionGe(cs.Properties.OrchestratorProfile.OrchestratorVersion, version)
 		},
-		"IsKubernetesVersionLt": func(version string) bool {
-			return cs.Properties.OrchestratorProfile.IsKubernetes() && !IsKubernetesVersionGe(cs.Properties.OrchestratorProfile.OrchestratorVersion, version)
```
Comment (xuto2) [inline pkg/agent/baker.go:220]: can we keep this?

File: pkg/agent/baker.go (hunk 1 above)
Comment (yizhang4321) [reply pkg/agent/baker.go:303]: Sure. "getVNETAddressPrefixes" removed.

File: pkg/agent/baker.go (hunk 2 above)
Comment (yizhang4321) [reply pkg/agent/baker.go:291]: getLBRules removed from utils.

File: pkg/agent/baker.go (hunk 3 above)
Comment (yizhang4321) [reply pkg/agent/baker.go:294]: getProbles removed from utils.go

File: pkg/agent/baker.go (hunk 5 above)
Comment (yizhang4321) [reply pkg/agent/baker.go:306]: getVNETSubnets removed from utils.go

File: pkg/agent/baker.go (hunk 6 above)
Comment (yizhang4321) [reply pkg/agent/baker.go:309]: getDataDisks removed from utils.go

File: pkg/agent/baker.go (hunk 7 above)
Comment (yizhang4321) [reply pkg/agent/baker.go:210]: This comment is on this line:
"IsKubernetesVersionGe": func(version string) bool {

but I didn't remove IsKubernetesVersionGe. Do you meant to comment it on a different function that I removed?

File: pkg/agent/baker.go (hunk 8 above)
Comment (yizhang4321) [reply pkg/agent/baker.go:220]: I added IsKubernetesVersionLt back per your comment.

File: pkg/agent/baker.go (hunk 4 above)
Comment (yizhang4321) [reply pkg/agent/baker.go:297]: getSecurityRules removed from utils.go

File: pkg/agent/utils.go [hunk 9]
```diff
@@ -250,17 +201,6 @@ func getLBRule(name string, port int) string {
...
-	for index, port := range ports {
-		if index > 0 {
-			buf.WriteString(",\n")
-		}
-		buf.WriteString(getLBRule(name, port))
-	}
-	return buf.String()
-}
-
 func getProbe(port int) string {
```
Comment (xuto2) [inline pkg/agent/utils.go:204]: this

Comment (yizhang4321) [reply pkg/agent/utils.go:204]: getProbe function is.being removed in this new PR: https://github.com/Azure/AgentBaker/pull/308

**Pull request merged**